From c21eb558c1f457d802017620ec4622d86be398ce Mon Sep 17 00:00:00 2001
From: Pavithra R <pavir@codeaurora.org>
Date: Fri, 10 Nov 2017 11:41:22 -0800
Subject: [PATCH 388/500] [tcrypt] AHASH  - Update test vetors for HMAC SHA1
 and HMAC SHA256.

Test vectors of key size larger than block is disabled for HMAC SHA1 & HMAC SHA256 respectively.

Change-Id: Idf1e5090731c064eadc02da82d1d841b69997c73
Signed-off-by: Pavithra R <pavir@codeaurora.org>
Signed-off-by: Tushar Ganatra <quic_tganatra@quicinc.com>
---
 crypto/testmgr.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/crypto/testmgr.h b/crypto/testmgr.h
index 39e7364d374a..789d69e69f88 100644
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -7288,7 +7288,10 @@ static const struct hash_testvec hmac_sha1_tv_template[] = {
 		.psize	= 20,
 		.digest	= "\x4c\x1a\x03\x42\x4b\x55\xe0\x7f\xe7\xf2"
 			  "\x7b\xe1\xd5\x8b\xb9\x32\x4a\x9a\x5a\x04",
-	}, {
+	},
+#ifndef CONFIG_CRYPTO_DISABLE_AHASH_LARGE_KEY_TEST
+	/* TODO : Require support for Auth key size larger than block length */
+	{
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -7316,6 +7319,7 @@ static const struct hash_testvec hmac_sha1_tv_template[] = {
 		.digest	= "\xe8\xe9\x9d\x0f\x45\x23\x7d\x78\x6d\x6b"
 			  "\xba\xa7\x96\x5c\x78\x08\xbb\xff\x1a\x91",
 	},
+#endif
 };
 
 
@@ -7538,7 +7542,9 @@ static const struct hash_testvec hmac_sha256_tv_template[] = {
 			  "\x1a\xb9\xc3\x74\x9a\x5f\x1c\x17"
 			  "\xd4\xf5\x89\x66\x8a\x58\x7b\x27"
 			  "\x00\xa9\xc9\x7c\x11\x93\xcf\x42",
-	}, {
+	},
+#ifndef CONFIG_CRYPTO_DISABLE_AHASH_LARGE_KEY_TEST
+	{
 		.key	= "\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
 			"\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa\xaa"
@@ -7570,6 +7576,7 @@ static const struct hash_testvec hmac_sha256_tv_template[] = {
 			  "\x84\xd3\xe7\xa1\xff\x98\xa2\xfc"
 			  "\x2a\xc7\xd8\xe0\x64\xc3\xb2\xe6",
 	},
+#endif
 };
 
 static const struct hash_testvec aes_cmac128_tv_template[] = {
-- 
2.34.1

